### ISSUE TITLE
Nagios cgi-bin error

### ISSUE THREAD

**Issue by keiner99** [chriscareycode/nagiostv-react#14]
Hello,
somethimes I got randomly this error:
![image](https://user-images.githubusercontent.com/9336808/74244112-393ef000-4ce1-11ea-96f2-9934257f0ea1.png)

but if I open the url http://nagiosurl/nagios/cgi-bin/statusjson.cgi?query=servicelist&details=true or http://nagiosurl/nagios/cgi-bin/statusjson.cgi?query=hostlist&details=true I can see all infos, there is no error.

Can you help me with this error? After restarting the browser everything is ok again

Best regards,
keiner99

**Comment by chriscareycode** [comment]
I wonder if since you have a lot of items that I need to relax the timeout on the request. It might be giving up too quickly

**Comment by keiner99** [comment]
yes maybe that's the problem. there are about 160 hosts and almost 500 services. can you increase the timeout in the next release?

**Comment by chriscareycode** [comment]
On those ajax calls there is no explicit timeout set so it was browser specific. We fetch hosts and services every 15s so I'll change the timeout to 13 seconds and let's see how that goes. I should be able to cut a release with this today or tomorrow.

**Comment by chriscareycode** [comment]
Published 0.4.10 https://github.com/chriscareycode/nagiostv-react/releases/tag/v0.4.10

**Comment by keiner99** [comment]
thx for your quick support! I installed the new version, let's see if the problem is gone now :)

**Comment by keiner99** [comment]
edit: I have the problem again.. 
![tempsnip](https://user-images.githubusercontent.com/9336808/75757102-0ad99100-5d32-11ea-9cf8-170b0ebb763c.png)

**Comment by chriscareycode** [comment]
Open your browsers Developer Tools -> Network -> XHR and leave that open. It will have some more debug information and also a "time" column to show how long the requests are taking. See if you can screenshot that before and after the error for more information

**Comment by keiner99** [comment]
sorry for the late response.. here are the XHR infos of the browser:

before the error:
![image](https://user-images.githubusercontent.com/9336808/77145146-04644c80-6a88-11ea-83ce-b1e1149c2311.png)

after the error:
![image](https://user-images.githubusercontent.com/9336808/77145175-16de8600-6a88-11ea-8712-38b2b199ba92.png)

I hope, that helps. 
Best regards, keiner99

**Comment by chriscareycode** [comment]
So bizarre. To rule out intermittent network issue: Can you also open a terminal from the box running the browser with the error, and run a constant ping to the nagios server. When NagiosTV has the failed requests are pings still steady?

**Comment by t2-eclectic** [comment]
same problem with servicelist, hosts working fine: nagiostv-0.5.3
query runs in a browser over quite a bit of time
returns a 2.8MB json file 

can you expose the timeout thru a setting?

**Comment by chriscareycode** [comment]
Yeah I can do that, and maybe just automatically set the polling interval to be a second or two longer than the timeout. So people with large JSON can experiment with longer timeout with slower poll, and others can do shorter timeout with faster poll. There are almost always going to be significantly larger servicelist than hostlist, so those should be different values.

**Comment by chriscareycode** [comment]
@t2-eclectic Still haven't forgotten about this one but I'm in the middle of a move so might be delayed a bit more. 

Right now the polling interval is at 15s and I believe the timeout on that is set to 14s. That seems more than enough time for 2.8MB to download. Is it taking longer than 14s when you try to load the json in the browser? 

Does it work sometimes and fail other times? Or is it always failing?

**Comment by chriscareycode** [comment]
I also get this from time to time. Just today I saw it on my NagiosTV. and my computer that the web browser is running on is hard wired to a switch, the same switch my Nagios server is connected to. So it would be a connectivity error between to machines plugged into the same switch. One thing I noticed though was I was loading up Amazon.com on this machine in another window, and the site was taking a long time to load, like 10 seconds. Strange for Amazon. At the same time, NagiosTV gave that cgi-bin error. I was thinking that is some temporary networking issue on my machine or the switch it's connected to.

Maybe a solution for now is to make some changes to the way the errors are presented to the NagiosTV screen. On many other web application, if a single api fetch fails, it might fail more silently and just retry. If NagiosTV has one bad fetch, maybe we should not pop up such a huge alarming CONNECTION ERROR. Right now it is configured that any single error fetching will pop up that error screen. 

We could fail once or twice a bit more silently, and just retry. If the retry works fine then that's no problem. It's really only when no connection can not be established over and over that we want to show a warning that something is not set up correctly or the server is completely down.
